tests/integration/suite/scheduler/metrics/basic.go

```diff
@@ -0,0 +1,187 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	schedulerv1 "github.com/dapr/dapr/pkg/proto/scheduler/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	frameworkclient "github.com/dapr/dapr/tests/integration/framework/client"
+	"github.com/dapr/dapr/tests/integration/framework/process/ports"
+	"github.com/dapr/dapr/tests/integration/framework/process/scheduler"
+	"github.com/dapr/dapr/tests/integration/suite"
+	"github.com/dapr/kit/ptr"
+)
+
+func init() {
+	suite.Register(new(basic))
+}
+
+type basic struct {
+	scheduler *scheduler.Scheduler
+
+	etcdPort   int
+	etcdClient *clientv3.Client
+	idPrefix   string
+}
+
+func (b *basic) Setup(t *testing.T) []framework.Option {
+	uuid, err := uuid.NewUUID()
+	require.NoError(t, err)
+	b.idPrefix = uuid.String()
+
+	fp := ports.Reserve(t, 2)
+	port1 := fp.Port(t)
+	port2 := fp.Port(t)
+
+	b.etcdPort = port2
+
+	clientPorts := []string{
+		"scheduler-0=" + strconv.Itoa(b.etcdPort),
+	}
+	b.scheduler = scheduler.New(t,
+		scheduler.WithID("scheduler-0"),
+		scheduler.WithInitialCluster(fmt.Sprintf("scheduler-0=http://localhost:%d", port1)),
+		scheduler.WithInitialClusterPorts(port1),
+		scheduler.WithEtcdClientPorts(clientPorts),
+	)
+
+	fp.Free(t)
+	return []framework.Option{
+		framework.WithProcesses(fp, b.scheduler),
+	}
+}
+
+func (b *basic) Run(t *testing.T, ctx context.Context) {
+	b.scheduler.WaitUntilRunning(t, ctx)
+
+	var err error
+	b.etcdClient, err = clientv3.New(clientv3.Config{
+		Endpoints:   []string{fmt.Sprintf("localhost:%d", b.etcdPort)},
+		DialTimeout: 5 * time.Second,
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, b.etcdClient.Close())
+	})
+
+	frameworkClient := frameworkclient.HTTP(t)
+	client := b.scheduler.Client(t, ctx)
+
+	t.Run("create 10 jobs, ensure metrics", func(t *testing.T) {
+		for i := 1; i <= 10; i++ {
+			name := b.idPrefix + "_" + strconv.Itoa(i)
+
+			req := &schedulerv1.ScheduleJobRequest{
+				Name: name,
+				Job: &schedulerv1.Job{
+					Schedule: ptr.Of("@every 20s"),
+					Repeats:  ptr.Of(uint32(1)),
+					Data: &anypb.Any{
+						Value: []byte(b.idPrefix),
+					},
+					Ttl: ptr.Of("30s"),
+				},
+				Metadata: &schedulerv1.JobMetadata{
+					AppId:     "appid",
+					Namespace: "namespace",
+					Target: &schedulerv1.JobTargetMetadata{
+						Type: new(schedulerv1.JobTargetMetadata_Job),
+					},
+				},
+			}
+
+			_, err := client.ScheduleJob(ctx, req)
+			require.NoError(t, err)
+
+			assert.True(t, b.etcdHasJob(t, ctx, name))
```

Comment:
Either use full path on the ETCD client Get for the job, or use the Scheduler gRPC client to get the Job.

```suggestion
```

Reply:
I think this is done?

Reply:
Please can we not just use the scheduler client to get the job so everything is fully qualified? I don't like the fact we are doing suffix checks with `||` to check for existence.